R/maraca.R

```diff
@@ -347,124 +430,109 @@ plot_maraca <- function(
     ) +
     ggplot2::guides(fill = "none")
 
+  # Add label to plot - maracaPlot
+  attr(plot, "class") <- c(attr(plot, "class"), "maracaPlot")
+
   return(plot)
 }
 
-#' Creates and returns the tte composite plot of the maraca data.
+#' Generic function to generate validation data for the maraca plot object.
 #'
-#' @param obj an object of S3 class 'maraca'
-#' @return a ggplot2 object of the data. This function
-#' will not render the plot immediately. You have to print() the returned
-#' object for it to be displayed.
+#' This will produce the 4 validation datasets.
+#'
+#' @param x an object of S3 class 'ggplot'
+#' @param \dots not used
+#' @return Creates a list of datasets for validation purposes.
 #'
 #' @examples
 #' data(hce_scenario_a)
 #' hce_test <- maraca(
 #'   data = hce_scenario_a,
 #'   tte_outcomes = c("Outcome I", "Outcome II", "Outcome III", "Outcome IV"),
 #'   continuous_outcome = "Continuous outcome",
+#'   fixed_followup_days = 3 * 365,
 #'   column_names = c(outcome = "GROUP", arm = "TRTP", value = "AVAL0"),
 #'   arm_levels = c(active = "Active", control = "Control"),
 #'   compute_win_odds = TRUE
 #' )
-#' plot <- plot_tte_composite(hce_test)
+#' p <- plot(hce_test)
+#' validate_maraca(p)
+#'
 #' @export
-plot_tte_composite <- function(obj) {
-  checkmate::assert_class(obj, "maraca")
+validate_maraca <- function(x,  ...) {
```

Comment:
This is a validation of the maraca plot, not of the maraca object. The naming is confusing and conflicts with the current assumed naming convention.

Also, can't you get the information from the maraca object itself, without creating the plot?